server/metadata.go

```diff
@@ -1323,3 +1309,39 @@ func ensureTimeout(ctx context.Context, defaultTimeout time.Duration) (context.C
 	}
 	return ctx, cancel
 }
+
+// eventTimestampsToProto returns a client proto's partition event timestamps
+// from a partition's event timestamps struct.
+func eventTimestampsToProto(timestamps eventTimestamps) *client.PartitionEventTimestamps {
+	first, latest := timestamps.firstTime, timestamps.lastTime
+	result := &client.PartitionEventTimestamps{}
+
+	// Calling UnixNano() on a zero time is undefined, so we need to make these
+	// checks.
+	if !first.IsZero() {
+		result.FirstTimestamp = first.UnixNano()
+	}
+	if !latest.IsZero() {
+		result.LatestTimestamp = latest.UnixNano()
+	}
+
+	return result
+}
+
+// getPartitionMetadata returns a partition's metadata.
+func getPartitionMetadata(partitionId int32, partition *partition) *client.PartitionMetadata {
+	leader, _ := partition.GetLeader()
+	return &client.PartitionMetadata{
+		Id:                 partitionId,
+		Leader:             leader,
+		Replicas:           partition.GetReplicas(),
+		Isr:                partition.GetISR(),
+		HighWatermark:      partition.log.HighWatermark(),
+		NewestOffset:       partition.log.NewestOffset(),
+		Paused:             partition.GetPaused(),
+		Readonly:           partition.GetReadonly(),
```

Comment:
It took me a moment to realize these are grabbing the _protobuf_ values, but I don't think we actually set those values when calling `SetReadonly`/`Pause` on the partition. The protobuf fields are mainly used for snapshotting of the Raft log, e.g. [here](https://github.com/liftbridge-io/liftbridge/blob/1478dcbc46ec50dcfb2527e084c683675d3e1fc8/server/fsm.go#L300-L301). That code in `Snapshot` could probably be removed if we just set the protobuf field in `SetReadonly` and `Pause`/`replacePartition`.

Reply:
Indeed, well spotted! Should be fixed in b1fa095.